File: QuSoC/QuSoC/apps/Counter/firmware/SOCCounter.cpp
#include "SOCCounter.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace SOCCounter
{
	void Firmware::EntryPoint()
	{
		uint32_t counter = 0;
		while(true)
		{
			(counter++);
			SOCCounter_SOC_Counter = counter;
		}
	}
}

File: QuSoC/QuSoC/apps/Counter/source/SOCCounter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SOCCounter
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual uint Counter { get; set; }
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            SOC.Instance.Counter = 100;

            /*
            uint counter = 0;
            while (counter < 10)
            {
                counter++;
                SOC.Instance.Counter = counter;
            }
            */
        }
    }
}
